text/chapter10.md

```diff
@@ -503,36 +503,56 @@ exports.unsafeHead = arr => {
 
 ## Using Type Class Member Functions
 
-Just like our earlier guide on passing the `Maybe` constructor over FFI, this is another case of writing PureScript that calls JavaScript, which in turn calls PureScript functions again. Here we will explore how to pass type class member functions over the FFI.
+Say we want to write a function that relies on its argument having a `Show` instance, in a foreign JavaScript module. We might want to do this:
 
-We start with writing a foreign JavaScript function which expects the appropriate instance of `show` to match the type of `x`.
+```hs
+-- Don't do this.
+
+foreign import boldBad :: forall a. Show a => a -> String
+```
+
+This would work provided the foreign module exports:
+
+```js
+// Seriously. Don't. Read on.
+
+exports.boldBad = Show => x =>
+  Show.show(x).toUppercase() + "!!!";
+```
+
+A constraint in a foreign import definition results in importing a curried function that accepts a _dictionary_ object containing the class members. The compiler implicitly passes, for us, a dictionary of the correct instance members when calling the foreign function. However, this practice has been DEPRECATED and doing so will trigger a warning. Use of dictionaries in providing access to type class members turned out to be more complex than anticipated.
```

Comment:
> turned out to be more complex than anticipated

I'm confused about where this was anticipated to be less complex. Are you talking about assumptions during language design, or when writing the previous example?

Reply:
Both. In a way, intentionally ambiguously. I don't want to bring up the details of how/why this is deprecated. So I use a blanket reason equivalent to "keeping it will be more trouble than dropping it."

Choices during language design results in writing more complex code. And also harder to write compilers. See [previous comment](https://github.com/purescript-contrib/purescript-book/pull/364#discussion_r650548786).

In the next paragraph I use the word "simple" to describe the proper way.

Reply:
I'm concerned that being vague here will be more confusing for readers.

Reply:
I agree, I think this last sentence raises more questions than it answers. If we are going to tell the reader not to do something, I think they probably deserve a slightly more satisfying explanation than this.

Reply:
Please see [comment](https://github.com/purescript-contrib/purescript-book/pull/364#discussion_r652603611).